### PR TITLE
Add pool transaction hashes access [ECR-3136]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Add pool transaction hashes access.

---
See: https://jira.bf.local/browse/ECR-3136

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java [hunk 1]
```diff
@@ -239,4 +240,11 @@ public Block getLastBlock() {
   public StoredConfiguration getActualConfiguration() {
     return schema.getActualConfiguration();
   }
+
+  /**
+   * Returns a set of uncommitted (in-pool) transaction hashes.
```
Comment (bullet-tooth) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java]: What will I get if mempool is empty, empty set or null?

File: exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/CoreSchemaProxy.java [hunk 2]
```diff
@@ -157,6 +158,14 @@ Block getLastBlock() {
         StandardSerializers.hash(), TRANSACTION_LOCATION_SERIALIZER);
   }
 
+  /**
+   * Returns a set of uncommitted (in-pool) transaction hashes.
+   */
+  KeySetIndexProxy<HashCode> getPoolTransactionHashes() {
```
Comment (bullet-tooth) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/CoreSchemaProxy.java]: add test to `CoreSchemaProxyIntegrationTest`

File: exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java [hunk 3]
```diff
@@ -239,4 +240,13 @@ public Block getLastBlock() {
   public StoredConfiguration getActualConfiguration() {
     return schema.getActualConfiguration();
   }
+
+  /**
+   * Returns a set of uncommitted (in-pool) transaction hashes; empty in case of no transactions.
+   *
+   * @see <a href="https://exonum.com/doc/version/0.10/advanced/consensus/specification/#pool-of-unconfirmed-transactions">Pool of Unconfirmed Transactions</a>
+   */
+  KeySetIndexProxy<HashCode> getPoolTransactionHashes() {
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java]: I am not entirely happy with the name, what if it is named `getTransactionPool` and the documentation clarifies that a transaction pool is a set of hashes of uncommitted transaction messages?

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java]: Is that correct that it is intended to be internal (hence package-private and, supposedly, invocation through reflection)? If it is not I would also add a warning that this pool represents the state as of the current snapshot, and its state constantly changes even between block commits.

Comment (MakarovS) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/blockchain/Blockchain.java]: Yes, I was inclining to make it public

**Pull request merged**